frontend/src/components/pongGame/pongGame.vue

```diff
@@ -248,41 +247,56 @@ export default defineComponent({
 
 		finishGame() {
 			const updateGameDto = {
-				score_player_one: score_player_one.value,
-				score_player_two: score_player_two.value,
+				score_player_one: this.score_player_one,
+				score_player_two: this.score_player_two,
 				state: 'done',
 			};
 			this.gameStatus = GameStatusEnum.GAME_OVER;
-			this.socket.off('updatePosition', this.render);
+			this.socket.disconnect();
+			// this.socket.off('updatePosition', this.render);
 			// patch game in database with the updated scores
 			patchRequest(`games/${this.currentgameId}`, updateGameDto);
 			const winner: string | undefined =
-				score_player_one > score_player_two
+				this.score_player_one > this.score_player_two
 					? this.getPlayerOne?.name
 					: this.getPlayerTwo?.name;
 			alert(`Game over!\nWell done ${winner}!`);
 		},
+
+		setSocketOn() {
+			this.socket.on('elementPositions', this.render);
+			this.socket.on(
+				'pointOver',
+				(scores: {
+					scorePlayerOne: number;
+					scorePlayerTwo: number;
+				}) => {
+					this.gameStatus = GameStatusEnum.POINT_OVER;
+					this.score_player_one = scores.scorePlayerOne;
+					this.score_player_two = scores.scorePlayerTwo;
+				},
+			);
+			this.socket.on('gameOver', () => {
+				router.push({ name: 'activeGames' });
+			});
+		},
 	},
 
 	mounted() {
 		this.context = (this.$refs.GameRef as any).getContext('2d');
-		document.addEventListener('keydown', this.keyDown);
-		document.addEventListener('keyup', this.keyUp);
-
-		this.socket.on('elementPositions', this.render);
-		this.socket.on(
-			'pointOver',
-			(scores: { scorePlayerOne: number; scorePlayerTwo: number }) => {
-				this.gameStatus = GameStatusEnum.POINT_OVER;
-				score_player_one.value = scores.scorePlayerOne;
-				score_player_two.value = scores.scorePlayerTwo;
-			},
-		);
-		this.socket.on('gameOver', () => {
-			router.push({ name: 'activeGames' });
-		});
+		console.log(this.getCurrentGame?.player_two.id,  this.getCurrentGame?.player_one.id,  this.getMyId);
+		if (this.isPlayer) {
+			console.log("\n\n\nsetting eventlisteners\n\n\n")
+			document.addEventListener('keydown', this.keyDown);
+			document.addEventListener('keyup', this.keyUp);
+		}
+		this.socket.emit('joinGameRoom', this.getCurrentGame);
+		this.setSocketOn();
+		
 		// MAIN GAME LOOP
-		this.startGameLoop();
+		this.socket.on("GameCanStart", () => {
+			this.startGameLoop();
+		})
```

Comment:
Since this should only happen once per player, you can use `this.socket.once(...`
that would create an event listener that automatically removes itself after it is called the first time